src/tribler/core/components/libtorrent/download_manager/download_manager.py

```diff
@@ -463,7 +463,7 @@ def update_ip_filter(self, lt_session, ip_addresses):
         lt_session.set_ip_filter(ip_filter)
 
     async def get_metainfo(self, infohash: bytes, timeout: float = 30, hops: Optional[int] = None,
-                           url: Optional[str] = None, raise_errors: bool = False) -> Optional[Dict]:
+                           url: Optional[Union[str, bytes]] = None, raise_errors: bool = False) -> Optional[Dict]:
```

Comment:
I realize this PR doesn't introduce this, so I'm not requesting you to change this, but it doesn't make a lot of sense to me for an url to be bytes.

Reply:
I also realized this after adding typings in this PR. I've created a related issue: https://github.com/Tribler/tribler/issues/7823